src/views/UrbitID.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { Grid } from 'indigo-react';
+
+import { useHistory } from 'store/history';
+
+import useCurrentPointName from 'lib/useCurrentPointName';
+import useRouter from 'lib/useRouter';
+import { LocalRouterProvider } from 'lib/LocalRouter';
+
+import View from 'components/View';
+import Crumbs from 'components/Crumbs';
+import NavHeader from 'components/NavHeader';
+
+import UrbitIDHome from './UrbitID/Home';
+import SigilGenerator from './UrbitID/SigilGenerator';
+import DownloadKeys from './UrbitID/DownloadKeys';
+import SetProxy from './UrbitID/SetProxy';
+import ResetKeys from './UrbitID/ResetKeys';
+import Transfer from './UrbitID/Transfer';
+
+const NAMES = {
+  HOME: 'HOME',
+  SIGIL_GENERATOR: 'SIGIL_GENERATOR',
+  DOWNLOAD_KEYS: 'DOWNLOAD_KEYS',
+  SET_PROXY: 'SET_PROXY',
+  RESET_KEYS: 'RESET_KEYS',
+  TRANSFER: 'TRANSFER',
+};
+
+const VIEWS = {
+  [NAMES.HOME]: UrbitIDHome,
+  [NAMES.SIGIL_GENERATOR]: SigilGenerator,
+  [NAMES.DOWNLOAD_KEYS]: DownloadKeys,
+  [NAMES.RESET_KEYS]: ResetKeys,
+  [NAMES.SET_PROXY]: SetProxy,
+  [NAMES.TRANSFER]: Transfer,
+};
+
+const humanizeName = name => {
+  switch (name) {
+    case NAMES.SIGIL_GENERATOR:
+      return 'Sigil';
+    case NAMES.DOWNLOAD_KEYS:
+      return 'Download Keys';
+    case NAMES.SET_PROXY:
+      return 'Set Key';
+    case NAMES.RESET_KEYS:
+      return 'Reset Keys';
+    case NAMES.TRANSFER:
+      return 'Transfer';
+    default:
+      return undefined;
+  }
+};
+
+export default function UrbitOS() {
+  const history = useHistory();
+  const name = useCurrentPointName();
+
+  const { Route, ...router } = useRouter({
+    names: NAMES,
+    views: VIEWS,
+    initialRoutes: [{ key: NAMES.HOME }],
+  });
+
+  const last = router.peek().key;
+  const homeAction = last === NAMES.HOME ? undefined : () => history.pop();
+  const lastCrumb = homeAction ? [{ text: humanizeName(last) }] : [];
+
+  return (
+    <View pop={router.pop} inset>
+      <NavHeader>
+        <Crumbs
+          routes={[
+            { text: name, action: () => history.pop() },
+            { text: 'Urbit ID', action: homeAction },
```

Comment:
It produces the below screens, which I prefer because you update the crumbs on every nav. Clearer for the user where they are.

<img width="962" alt="Screen Shot 2020-03-06 at 1 10 45 pm" src="https://user-images.githubusercontent.com/46801558/76046624-1df79580-5fac-11ea-9371-d8192d29e3e8.png">


<img width="946" alt="Screen Shot 2020-03-06 at 1 10 51 pm" src="https://user-images.githubusercontent.com/46801558/76046617-1afca500-5fac-11ea-94a8-4ec299b93cf8.png">

Reply:
I get the visual structure of the crumbs. But I'm reading/experiencing this that clicking "~name" is equivalent to clicking the back button, and clicking "Urbit ID" takes you to the point overview page.

Reply:
I think this still stands? Or am I just still confused about this?